### PR TITLE
Loading artist data from firebase

### PR DESCRIPTION
**Pull request by jamesmorgan**
* [x] Populate artists data and make the app work
  * [x] Fix gallery
  * [x] Fix artists overview page
  * [x] Fix artist page
* [x] Allow artists to customise details
	* [x] API https://github.com/knownorigin/known-origin-serverless-api/pull/9 
* [x] Auto refresh artist data when changed
* [x] Attempt to improve image shuffle issue
* [ ] Attempt to remove old/new eth addresses (only one per artists)?

### CODE REVIEW

File: src/components/ui-controls/artist/ArtistDataControlPanel.vue [hunk 1]
```diff
@@ -0,0 +1,279 @@
+<template>
+  <div class="container-fluid d-none d-md-block">
```
Comment (andygray) [inline src/components/ui-controls/artist/ArtistDataControlPanel.vue]: 👍 not on mobile - nice

File: src/components/ui-controls/artist/ArtistDataControlPanel.vue [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+          </div>
+          <div class="col">
+            <div class="form-group">
+              <label for="twitterInput">Twitter</label>
+              <input type="text" class="form-control" id="twitterInput" placeholder="@" minlength="3" maxlength="100"
+                     v-model="form.twitter"
+                     @input="validateTwitter"
+                     :class="{ 'is-invalid': errors.twitter }">
+              <div class="invalid-feedback">
+                Invalid twitter handle
```
Comment (andygray) [inline src/components/ui-controls/artist/ArtistDataControlPanel.vue]: `handle` is a bit geeky term? `username`?

File: src/components/ui-controls/artist/ArtistDataControlPanel.vue [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+        </button>
+
+        <span v-if="state.triggered" class="pl-2 small text-info">
+          Sending update ...
+        </span>
+        <span v-if="state.success" class="pl-2 small text-success">
+          Changes applied
+        </span>
+        <span v-if="state.failure" class="pl-2 small text-warning">
+          Failed - please contact the KO team
```
Comment (andygray) [inline src/components/ui-controls/artist/ArtistDataControlPanel.vue]: use same message as auctions that have telegram link too?

File: src/App.vue [hunk 4]
```diff
@@ -169,7 +169,7 @@
       } else {
         console.log('Running without a web3 provider - falling back to infura');
 
-        window.web3 = new Web3(new Web3.providers.HttpProvider('https://mainnet.infura.io/nbCbdzC6IG9CF6hmvAVQ'));
+        window.web3 = new Web3(new Web3.providers.HttpProvider('https://mainnet.infura.io/v3/nbCbdzC6IG9CF6hmvAVQ'));
```
[the remark this answers is not in the document]
Comment (jamesmorgan) [reply src/App.vue]: ye infura changed to `v3`

File: src/components/pages/ArtistPage.vue [hunk 5]
```diff
@@ -96,41 +105,46 @@
...
-    },
-    methods: {
-      lookupArtist: function () {
+      editions() {
+        return this.editionsForArtist(this.artistAddress);
+      },
+      artist() {
         return this.findArtistsForAddress(this.$route.params.artistAccount);
       },
+      artistAddress() {
```
[the remark this answers is not in the document]
Comment (jamesmorgan) [reply src/components/pages/ArtistPage.vue]: ye - think we attack this once we have an idea of what to do with multiple addresses

File: src/components/ui-controls/artist/ArtistDataControlPanel.vue (hunk 2 above)
Comment (jamesmorgan) [reply src/components/ui-controls/artist/ArtistDataControlPanel.vue]: good call

**Pull request merged**